src/stavisky_lab_to_nwb/redis_interfaces/redisrecordingextractor.py

```diff
@@ -0,0 +1,276 @@
+            `RedisExtractorMixin`
+        sampling_frequency : float, optional
+            The sampling frequency of the data in Hz. See
+            `RedisExtractorMixin`
+        timestamp_source : bytes or str, optional
+            The source of the timestamp information in the Redis
+            stream. See `RedisExtractorMixin`
+        timestamp_kwargs : dict, default: {}
+            If timestamp source is not "redis", then timestamp_kwargs
+            must contain the unit, dtype, and encoding of the timestamp
+            data, and optionally smoothing parameters. See
+            `RedisExtractorMixin.get_ids_and_timestamps()`
+        gain_to_uv : float, optional
+            Scaling necessary to convert the recording values to
+            microvolts
+        channel_dim : int, default: 0
+            If frames_per_entry > 1, then channel_dim indicates the
+            axis ordering of the data in each entry. If channel_dim
+            = 0, then the data is assumed to originally be of shape
+            (channel_count, frames_per_entry). If channel_dim = 1,
+            then the data is assumed to originally be of shape
+            (frames_per_entry, channel_count)
+        """
+        # Instantiate Redis client and check connection
+        self._client = redis.Redis(
+            port=port,
+            host=host,
+        )
+        self._client.ping()
+
+        # Construct channel IDs if not provided
+        if channel_ids is None:
+            channel_ids = np.arange(channel_count, dtype=int).tolist()
+
+        # get entry IDs and timestamps
+        stream_len = self._client.xlen(stream_name)
+        assert stream_len > 0, "Stream has length 0"
+        start_time, sampling_frequency, timestamps, entry_ids = self.get_ids_and_timestamps(
+            stream_name=stream_name,
+            frames_per_entry=frames_per_entry,
+            start_time=start_time,
+            sampling_frequency=sampling_frequency,
+            timestamp_source=timestamp_source,
+            **timestamp_kwargs,
+        )
+
+        # Initialize Recording and RecordingSegment
+        # NOTE: does not support multiple segments, assumes continuous recording for whole stream
+        BaseRecording.__init__(self, channel_ids=channel_ids, sampling_frequency=sampling_frequency, dtype=dtype)
+        recording_segment = RedisStreamRecordingSegment(
+            client=self._client,
+            stream_name=stream_name,
+            data_key=data_key,
+            dtype=dtype,
+            channel_count=channel_count,
+            timestamps=timestamps,
+            entry_ids=entry_ids,
+            frames_per_entry=frames_per_entry,
+            t_start=None,  # t_start != start_time
+            channel_dim=channel_dim,
+        )
+        self.add_recording_segment(recording_segment)
+
+        # Set gains if provided
+        if gain_to_uv is not None:
+            self.set_channel_gains(gain_to_uv)
+
+        # Not sure what this is for?
```

Comment:
Believe it's for pickling primarily (dumping the extractor object to a serialized form so it can be reloaded elsewhere)

Reply:
Would be the kind of thing they would comment on if/when we contribute this upstream